pilot/pkg/model/sidecar.go

```diff
@@ -556,3 +520,44 @@ func (ilw *IstioEgressListenerWrapper) selectServices(services []*Service, confi
 	}
 	return filteredServices
 }
+
+func matchingServices(importedHosts []host.Name, service *Service, ilw *IstioEgressListenerWrapper) (bool, []*Service) {
```

Comment:
Cool,  i like this

Reply:
but can we remove the bool return value?

Reply:
We can remove, it will be empty service if nothing is matched. I will do it next commit.